src/main/java/com/google/devtools/build/lib/runtime/commands/CleanCommand.java

```diff
@@ -178,8 +201,12 @@ private void actuallyClean(CommandEnvironment env,
           env.getWorkspaceName(), "execroot/" + env.getWorkspaceName() }) {
         Path child = outputBase.getRelative(directory);
         if (child.exists()) {
-          LOG.finest("Cleaning " + child);
-          FileSystemUtils.deleteTreesBelow(child);
+          LOG.finest("Cleaning " + child + (cleanOptions.async ? " asynchronously..." : ""));
+          if (cleanOptions.async) {
+            asyncClean(env, child, "Child");
```

Comment:
Can you make the string a bit more descriptive? Lukacs suggested 'workspace_root'.